app/forms/decidim/participatory_processes/admin/participatory_process_form.rb

```diff
@@ -93,8 +99,63 @@ def processes
           @processes ||= Decidim::ParticipatoryProcess.where(organization: current_organization)
         end
 
+        def remove_emitter_image=(value)
+          self.remove_emitter = value if value
+          prepare_emitter_name
+        end
+
+        def emitter_name_image=(value)
+          @emitter_name_image = value if value
+          prepare_emitter_name
+        end
+
+        def emitter_name_select=(value)
+          @emitter_name_select = value if value
+          prepare_emitter_name
+        end
+
+        def emitter_image=(value)
+          @emitter_image = value if value
+          prepare_emitter
+        end
+
+        def emitter_select=(value)
+          blob = Decidim::ParticipatoryProcess.find(value).emitter.attachment.blob if value.present?
+          @emitter_select = blob if blob
+          @emitter_name_select = Decidim::ParticipatoryProcess.find(value).emitter_name if value.present?
+          prepare_emitter
+          prepare_emitter_name
+        end
+
+        def emitter_image
+          @emitter_image || emitter
+        end
+
+        def emitter_select
+          @emitter_select || emitter
+        end
+
+        def emitter_name_image
+          @emitter_name_image || emitter_name
+        end
+
+        def emitter_name_select
+          @emitter_name_select || emitter_name
+        end
+
         private
 
+        def prepare_emitter_name
```

Comment:
Since this function updates `self.emitter_name` I think it can be easier to read if you create an alias `prepare_emitter_name!`. It will help to understand that it modifies directly the variable